### PR TITLE
Change recommendations to prefer Bash on Windows

### PR DESCRIPTION
**Pull request by samhutchins**
Rework the README to recommend using Bash on Ubuntu on Windows,
and pull everything into one README

Is there any information that's been lost that you want to preserve?

Can you also check my spelling/typos. I invariably read what I meant to write, not what's written.

### CODE REVIEW

**Comment by JMoVS** [comment]
I will take some days. I think when I'm in Cusco, I should find the minute to review it (I'm currently traveling through South America)

File: README.md [hunk 1]
```diff
@@ -1,18 +1,115 @@
...
-- [installing it in the linux subsystem for windows](https://github.com/JMoVS/installing_video_transcoding_on_windows/blob/master/lxss_method.md)
+### Windows 10
+If you're using an up-to-date version of Windows 10, you have access to "[Bash on Ubuntu on Windows](https://msdn.microsoft.com/en-gb/commandline/wsl/about)". This feature allows you to run command line Linux programs on Windows without using a virtual machine, and it also gives us access to Ubuntu's extensive package archives. Everything we need to run these scripts is easily installable with apt-get.
 
-## Notes about the two methods:
+1. First up, you need to activate Bash.
+    1. Open Settings then go to `Update & Security -> For Developers`
+    1. Change your computer to "Developer Mode"
+    1. Click `Start` then look for `Turn Windows features on or off`
+    1. Select `Windows Subsystem for Linux (beta)`
```
Comment (JMoVS) [inline README.md:12]: Did you test it this way? Is it not needed to go to the Windows Store and actually click the "download" button on Ubuntu for Windows?

File: README.md [hunk 2]
```diff
@@ -1,18 +1,115 @@
...
+    1. Click ok. You'll need to restart your PC
+    1. Now open up `cmd.exe` and run `bash`. Follow the instructions then bash is set up and ready to go 
+1. Next we install the dependencies for `video_transcoding`
+    1. Fire up bash
```
Comment (JMoVS) [inline README.md]: Firing up bash can be quite difficult as sometimes, Windows didn't show it in search and the icon on the desktop only appearts after you accept the license agreement

File: README.md [hunk 3]
```diff
@@ -1,18 +1,115 @@
...
+    1. Run `sudo apt-get update && sudo apt-get upgrade` to ensure everything's up to date
```
Comment (JMoVS) [inline README.md]: Also, the first time you run bash, it asks you to create a user and a password, should we mention that here as well?

File: README.md [hunk 4]
```diff
@@ -1,18 +1,115 @@
...
+    1. Run `sudo apt-get install ruby ffmpeg mp4v2-utils mkvtoolnix`
+    1. Install HandBrakeCLI
+        1. `sudo add-apt-repository ppa:stebbins/handbrake-releases`
+        1. `sudo apt-get update`
+        1. `sudo apt-get install handbrake-cli`
+1. Now we can install `video_transcoding`
+    1. `sudo gem install video_transcoding`
 
-### Manually installing binaries
+The `detect-crop` command has an optional dependency on `mpv` for previewing crops. You can't launch graphical Linux apps using "Bash on Ubuntu on Windows", but it is capable of launching Windows programs; thos means we can install `mpv` for Windows then use an alias to launch it from bash:
```
Comment (JMoVS) [inline README.md]: I've heard that one could install a window manager like kde, so thereby possibly also launching mpv? :P

Also "this" means.

Comment (JMoVS) [reply README.md]: "this" means we can install `mpv` for Windows (comma or "and")

File: README.md [hunk 5]
```diff
@@ -1,18 +1,115 @@
...
 
-This option is recommended for most users. You will get a working system with all the binaries in place that will do the job. Updating the dependencies isn't automated though; you'll need to go back to each project's respective download page again and manually check if there's an update. As the dependencies aren't updated that often, this shouldn't really be a problem.
+1. Create `C:\bin` and [add it to your path](#adding-a-folder-to-your-path)
+1. Download a Windows build of [mpv](https://mpv.io/) from [here](https://mpv.srsfckn.biz/)
+1. Extract it to `C:\bin`
+1. Create an alias for mpv:
+    `echo "alias mpv=mpv.exe" >> .bashrc`
 
-### Installing it in the linux subsystem for windows
+If you have your movie files in the Windows file system (so NOT in `~userfolder/AppData/Local/lxss/home`), for example on your desktop, you have to provide the path to it via `/mnt/[drive letter]/path/to/your/file/location`. You can also navigate to the folder in Explorer, then type bash in the address bar to open a bash instance to that location
```
Comment (JMoVS) [inline README.md:34]: oh cool, didn't know that trick with bash in the address bar!

File: README.md [hunk 6]
```diff
@@ -1,18 +1,115 @@
...
 
-This method is bleeding edge as in: it is quite cumbersome and very experimental right now. The whole subsystem is in beta status (and will be for the foreseeable future). The spring Windows 10 update is supposed to fix some issues with it so that the need to self-compile everything will be gone, further streamlining and simplifying the process.
-This guide is for the adventurous. It will result in a system that you can easily update (as detailed in the guide) but is hard to set up right now. It is also unclear whether it breaks with an update easily. Not really recommended until the spring update has landed for wide availability.
+### Pre Windows 10 Creators Update and earlier
+In old versions of Windows 10, or in Windows 8.1 and earlier you either  don't have access to Bash on Ubuntu on Windows, or the version that's there can't be used to do this. But no matter! All the dependencies are available for Windows, you just need to track them down:
```
Comment (JMoVS) [inline README.md]: Is "But no matter!" a colloquialism that works in English?

File: README.md [hunk 7]
```diff
@@ -1,18 +1,115 @@
...
+        for /F "tokens=* USEBACKQ" %%F in (`type "!crop_file!"`) do set crop_option=--crop %%F
+    ) else (
+        set crop_option=
+    )
+
+    call transcode-video !crop_option! "%%I"
+)
+```
+
+It works in much the same way that Don's [bash script](https://github.com/donmelton/video_transcoding#batch-control-for-transcode-video) works, although it does have the limitation of not being able to "resume" the queue in the way that Don's does. I would recommend creating a `batch-transcode.bat` file in your `C:\bin` folder, then you can create the necesarry file structure anywhere:
```
Comment (JMoVS) [inline README.md:106]: I didn't look into the batch script that much to be honest, but what is limiting the resuming functionality?

**Review by JMoVS — CHANGES_REQUESTED** [review]
Very nice new document! I have some remaining questions but all in all I think this will improve usability a lot!

File: README.md (hunk 1 above)
Comment (samhutchins) [reply README.md:12]: It's not in the store. I think it's planned, but it's not there yet. Not in the UK at least.

File: README.md (hunk 2 above)
Comment (samhutchins) [reply README.md]: By this point people have run bash from CMD and got it all set up, including accepting license agreements. It should be visible in the start menu. How would you recommend starting bash?

File: README.md (hunk 3 above)
Comment (samhutchins) [reply README.md]: I'll add a note about that

File: README.md (hunk 4 above)
Comment (samhutchins) [reply README.md]: It's possible, I've got an X server in Windows that lets me use `git-gui` and `meld`. I can test mpv, but I don't have high hopes about it's performance

File: README.md (hunk 5 above)
Comment (samhutchins) [reply README.md:34]: It's a damn useful trick. Works with cmd and powershell too

File: README.md (hunk 6 above)
Comment (samhutchins) [reply README.md]: Yeah, informal English. I'll remove it, I don't think it quite works

File: README.md (hunk 7 above)
Comment (samhutchins) [reply README.md:106]: Don's shell script modifies the queue.txt to remove items it's already encoded, but I'm not aware of an easy built in way to do that in a batch file

File: README.md [hunk 8]
```diff
@@ -1,18 +1,115 @@
...
+## Transcoding video on Windows, the Don Melton way
 
+Don Melton provides some [amazing scripts](https://github.com/donmelton/video_transcoding) that wrap [HandBrake](https://handbrake.fr/) to produce high quality, portable versions of Blu Ray rips. The tools Don provides are written in [Ruby](https://www.ruby-lang.org/en/), and they're packaged as a [Ruby Gem](https://rubygems.org/). All the tools are cross platform, requiring only the Ruby runtime to be installed. But they do depend on external programs to get anything useful done. These dependencies are cross platform as well, but Windows has no standard package management system (de facto or otherwise), so installing these dependencies for Windows can be a bit of a pain. But it is possible! There are several ways to go about installing these tools and their dependencies. Downloading binaries manually, or using [Bash on Ubuntu on Windows](https://msdn.microsoft.com/en-gb/commandline/wsl/about) are two such options
```
Comment (JMoVS) [inline README.md]: I just noticed one more request, sorry for not including it earlier. I think we should reverse the order of "hunting down binaries or lxss method". For me the first option sounds like the suggested/recommended one (even though it isn't anymore). 

Maybe something like
"Using [Bash on Ubuntu on Windows](https://msdn.microsoft.com/en-gb/commandline/wsl/about) (default and recommended) or downloading binaries manually (for older Windows versions) are two such that are covered here."

File: README.md (hunk 1 above)
Comment (JMoVS) [reply README.md:12]: Hmm, Even if you search for Ubuntu? Strange, anyway, you tested it and it worked so 👍

File: README.md (hunk 8 above)
Comment (JMoVS) [reply README.md]: ... BlueRay and DVD rips.

File: README.md [hunk 9]
```diff
@@ -1,18 +1,115 @@
...
 
-## The two methods:
-- [manually installing the binaries](https://github.com/JMoVS/installing_video_transcoding_on_windows/blob/master/native_method.md)
-- [installing it in the linux subsystem for windows](https://github.com/JMoVS/installing_video_transcoding_on_windows/blob/master/lxss_method.md)
+### Windows 10
+If you're using an up-to-date version of Windows 10, you have access to "[Bash on Ubuntu on Windows](https://msdn.microsoft.com/en-gb/commandline/wsl/about)". This feature allows you to run command line Linux programs on Windows without using a virtual machine, and it also gives us access to Ubuntu's extensive package archives. Everything we need to run these scripts is easily installable with apt-get.
```
Comment (JMoVS) [inline README.md]: I don't know how perfectionistic we want to go about this (in all seriousness), a small thing is that it changes person from "you have access to Bash..." to "gives *us* access to..."

File: README.md [hunk 10]
```diff
@@ -1,18 +1,115 @@
-# Transcoding video on Windows, the Don Melton way
-Don Melton provides some [amazing scripts](https://github.com/donmelton/video_transcoding) that wrap [HandBrake](https://handbrake.fr/) to produce high quality, portable versions of Blu Ray rips. The tools Don provides are written in [Ruby](https://www.ruby-lang.org/en/), and they're packaged as a [Ruby Gem](https://rubygems.org/). All the tools are cross platform, requiring only the Ruby runtime to be installed. But they do depend on external programs to get anything useful done. These dependencies are cross platform as well, but Windows has no standard package management system (de facto or otherwise), so installing these dependencies for Windows can be a bit of a pain. But it is possible! There are several ways to go about installing these tools and their dependencies. Downloading binaries manually, or using the Linux Subsystem for Windows are two such options
+## Transcoding video on Windows, the Don Melton way
```
Comment (JMoVS) [inline README.md:1]: @donmelton Checking with you if you are fine with us naming this document this way? Is this ok for you? It implies an implcit approval of the outlined method from you, so checking with you in advance.

File: README.md (hunk 9 above)
Comment (JMoVS) [reply README.md]: Also from addressing the reader as "you", it changes to "everything *we* need to run"

File: README.md (hunk 5 above)
Comment (JMoVS) [reply README.md:34]: Super pedantic little things: Are there spaces after the `s ? Do you think it looks better with a soace? I trust your judgement there.

File: README.md [hunk 11]
```diff
@@ -1,18 +1,115 @@
...
+- [HandBrakeCLI](https://handbrake.fr/downloads2.php)  
+    Get the latest stable version
+- [ffmpeg](https://ffmpeg.org/download.html), the download page for Windows is [here](http://ffmpeg.zeranoe.com/builds/)  
+   Get the latest stable 'static' release.
+- [mkvtoolnix](https://mkvtoolnix.download/downloads.html#windows), the download page for Windows is [here](https://www.fosshub.com/MKVToolNix.html)  
+    Download the latest portable version. It's packaged as a .7z file, I'd recommend using [7zip](http://www.7-zip.org/download.html) to open these files
+- [mpv](https://mpv.io/), the download page for Windows is [here](https://mpv.srsfckn.biz/)  
+    Get the latest version
+- [MP4v2](https://code.google.com/archive/p/mp4v2/)  
+    Functional Windows binaries for this are hard to find, I've been using [this one](http://forum.doom9.org/showthread.php?t=171038). If anyone knows of a more official binary, please let me know
```
Comment (JMoVS) [inline README.md]: I think we should make the "please let me know" either a link to a new bug report or ot your email/twitter, so people could actually figure out quickly who,pm to contact (you).

**Review by JMoVS — CHANGES_REQUESTED** [review]
I went over it completely and from my side, this is really good! Do we want to open a separate pull request about how to update the different setups or do we want to include a "how to update" section in the additional info in this PR already?

File: README.md (hunk 8 above)
Comment (JMoVS) [reply README.md]: Blue-ray

File: README.md (hunk 5 above)
Comment (samhutchins) [reply README.md:34]: No spaces

File: README.md (hunk 8 above)
Comment (samhutchins) [reply README.md]: Blu-ray is the correct spelling

File: README.md (hunk 9 above)
Comment (samhutchins) [reply README.md]: I'll use you consistently

**Review by JMoVS — APPROVED** [review]

**Comment by JMoVS** [comment]
I'm gona squash and merge this now. I don't care enough about commit message typos ;-)

**Comment by JMoVS** [comment]
managed to clean up the commit messages, merging now.

**Comment by JMoVS** [comment]
Turns out, I would have been able to modify the commit message on the webinterface as well - well...

**Pull request merged**